src/magic.cpp

```diff
@@ -835,6 +864,30 @@ std::string spell::energy_cur_string( const player &p ) const
         const std::pair<std::string, nc_color> pair = p.get_fatigue_description();
         return colorize( pair.first, pair.second );
     }
+    if ( energy_source() == item_energy ) {
+        const itype_id id = energy_item();
+        const inventory& player_inv = p.inv;
+        std::vector<const item*> a_filter = player_inv.items_with( [id] (const item& it ) {
+            return it.typeId() == id;
+        } );
+        int count = 0;
+        for ( const item * it : a_filter ) {
+            if ( it->count_by_charges() ) {
+                count += p.inv.charges_of( energy_item() );
+            }
+            else {
+                count++;
+            }
+        }
+        count /= energy_cost( p );
```

Comment:
Might want to check if `energy_cost` is 0 here to avoid division by zero.

Reply:
Added the check up at the top, since there's no point in doing the rest of it if it's 0 anyways.